Docs 9551 - PM Admin Topics Review - ESB (#107)
mulesoft/docs-partner-manager: modules/ROOT/pages/upgrade-message-flows.adoc

> = Template Version Upgrades

Reviewer: What templates? When? Why?
Consider something more specific and transitional from previous texts? I don't know if this is accurate, but something like "Improved B2B Transactions Through Runtime Template Upgrades"?

> Partner Manager leverages prebuilt templates that convert your B2B message flow configurations into runtime applications. These templates are periodically updated to add new features and enhancements. You can upgrade your message flows to use the updated templates and to take advantage of the new capabilities that enable better integration with your trading partner's ecosystem.

Reviewer: "leverages" always makes me wonder "from what source," so I avoid it if a source is not named.
```suggestion
Partner Manager uses prebuilt templates to convert your B2B message flow configurations into runtime applications. As these templates are updated, you can upgrade your message flows to use them to take advantage of new capabilities that enable better integration with your trading partner's ecosystem.
```
The text in this document uses SO MANY different versions of naming for features. As you become more familiar with the product, I hope you can figure out how to distill them to one word or phrase apiece. For example, "your B2B message flow configurations": do we use B2B, or do we belabor you + partner or your organization + your partner? Are these messages or flows or transmissions or configurations? Makes my head spin. ;-)

Runtime templates related to Anypoint Partner Manager 2.5 are upgraded when:

* A new feature or functionality is added

> * A dependent component, such as a connector version, is upgraded

Reviewer: Is "dependent component" a clear or precise term for this?

> == Perform the Upgrade

Reviewer: As this is an imperative title, I expected steps. 
Not convinced that we need a subtopic here. 
How about going straight into 
"When a new template is available, the blah button in this location is activated. Before you click it, know that this can take some time and that it is safest for you to first upgrade in a nonproduction environment and test the new template before activating it in production."
The narrative is a little choppy as-is.

Reviewer: Proofread below

When new template versions are released, you see the *Upgrade deployments* button enabled in the top right corner of the *Message Flows* page.

To take advantage of new features, click the *Upgrade deployments* button. The process of updating all your message flows to use the latest template version begins.

> This process can take several minutes, depending on the number of message flow configurations and runtime applications in your environment.

Reviewer: To take advantage of new features, click the *Upgrade deployments* button. The process of updating all your message flows can take several minutes, depending on the number of message flow configurations and runtime applications in your environment.

When your message flows are deployed to CloudHub, your existing message flows are uninterrupted during the upgrade process.

When your message flows are deployed to your hosted runtimes, your runtime applications are stopped when the upgrade process begins, and then restarted when the upgrade is complete, which results in down time.

> It is best to perform the upgrade in your non-production environments and then to test some of your message flows before you perform the upgrade in your production environment.

Reviewer: ```suggestion
Upgrade your nonproduction environments first to test the changes on some of your message flows before upgrading your production environment.
```
